refactor!: update `math/base/special/cflipsignf` to follow latest project conventions

vivekmaurya001 opened the pull request:
## Description

- refactors C implementation for @stdlib/math/base/special/cflipsignf along with relevant tests, benchmarks and examples to follow latest conventions
- added test.native.js,  benchmark.native.js, addon file , native.js
- updated the readme file , package.json , benchmark.c , examples.c
- updated the initial test file to use absf instead of abs

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

 - [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)


* * *

@stdlib-js/reviewers 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

vivekmaurya001 committed:
feat(add c implementation): add math/base/special/cflipsignf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: failed
---

vivekmaurya001 committed:
feat(add c implementation): add math/base/special/cflipsignf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

vivekmaurya001 committed:
feat(add c implementation): add math/base/special/cflipsignf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

vivekmaurya001 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

vivekmaurya001 committed:
feat(add c implementation): add math/base/special/cflipsignf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

vivekmaurya001 committed:
Merge branch 'math/cflipsignf' of https://github.com/vivekmaurya001/stdlib into math/cflipsignf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

vivekmaurya001 committed:
feat(add c implementation): add math/base/special/cflipsignf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

vivekmaurya001 committed:
feat(add c implementation): add math/base/special/cflipsignf

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

vivekmaurya001 commented:
good morning @kgryte  @Planeshifter , @gunjjoshi   , it is ready for review

gunjjoshi commented:
@vivekmaurya001 Looks like there is a conflict in `example.c`. Can you resolve it?

vivekmaurya001 committed:
changed files

vivekmaurya001 commented:
@gunjjoshi  , resolved the conflicts

anandkaranubc commented:
@vivekmaurya001 Could you please resolve the conflict in `benchmark.c` and review the changes in https://github.com/stdlib-js/stdlib/pull/4767 to see if they are applicable here as well?

Thanks!

vivekmaurya001 committed:
Merge branch 'develop' into math/cflipsignf

Signed-off-by: Vivek maurya <155618190+vivekmaurya001@users.noreply.github.com>

vivekmaurya001 commented:
@anandkaranubc , I have resolved the conflicts , thanks for reviewing !

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/cflipsignf/src/main.c:
```diff
@@ -28,15 +29,29 @@
+* stdlib_complex64_t out = stdlib_base_cflipsignf( z, -1.0 );
+*
+* float re = stdlib_complex64_real( out );
+* // returns -3.5
```
```suggestion
* // returns -3.5f
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/cflipsignf/src/main.c:
```diff
@@ -28,15 +29,29 @@
+* // returns -3.5
+*
+* float im = stdlib_complex64_imag( out );
+* // returns 2.5
```
```suggestion
* // returns 2.5f
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/cflipsignf/src/main.c:
```diff
@@ -28,15 +29,29 @@
-	float re = crealf( z );
-	float im = cimagf( z );
+stdlib_complex64_t stdlib_base_cflipsignf( const stdlib_complex64_t z, const float y ) {
+	float re ;
```
```suggestion
	float re;
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/cflipsignf/src/main.c:
```diff
@@ -28,15 +29,29 @@
-	float im = cimagf( z );
+stdlib_complex64_t stdlib_base_cflipsignf( const stdlib_complex64_t z, const float y ) {
+	float re ;
+	float im ;
```
```suggestion
	float im;
```

anandkaranubc commented:
@vivekmaurya001 Following up on my previous comment, could you ensure that the changes made in https://github.com/stdlib-js/stdlib/pull/4767 are also reflected here? I still see the same pending changes here.

vivekmaurya001 committed:
Merge branch 'stdlib-js:develop' into math/cflipsignf

vivekmaurya001 committed:
chore: corrected some linting

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

vivekmaurya001 committed:
chore: corrected some linting

anandkaranubc commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into math/cflipsignf

anandkaranubc committed:
chore: apply suggestions from review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: remove line break

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM. Thanks, @gunjjoshi and @anandkaranubc, for the reviews!
